### PR TITLE
Resume provider support and Deprecate junos_scp and junos_package modules

### PR DESCRIPTION
**Pull request by rohitthakur2590**
Signed-off-by: rohitthakur2590 <rohitthakur2590@outlook.com>

##### ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- New Module Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: plugins/doc_fragments/junos.py [hunk 1]
```diff
@@ -11,7 +11,65 @@
 class ModuleDocFragment(object):
 
     # Standard files documentation fragment
-    DOCUMENTATION = r"""options: {}
+    DOCUMENTATION = r"""options:
```
Comment (Qalthos) [inline plugins/doc_fragments/junos.py:14]: We absolutely do not want to return this to every module. Just package and scp should be fine.

File: plugins/modules/junos_package.py [hunk 2]
```diff
@@ -13,7 +13,7 @@
 DOCUMENTATION = """
 module: junos_package
 author: Peter Sprygada (@privateip)
-short_description: Installs packages on remote devices running Junos
+short_description: (deprecated, removed after 2025-12-01) Installs packages on remote devices running Junos
```
Comment (Qalthos) [inline plugins/modules/junos_package.py:16]: Are we really deprecating this one though? It should work without provider even if it doesn't technically use the connection plugin
```suggestion
short_description: (deprecated, removed after 2025-01-01) Installs packages on remote devices running Junos
```

File: plugins/modules/junos_scp.py [hunk 3]
```diff
@@ -13,10 +13,14 @@
 DOCUMENTATION = """
 module: junos_scp
 author: Christian Giese (@GIC-de)
-short_description: Transfer files from or to remote devices running Junos
+short_description: (deprecated, removed after 2025-12-01) Transfer files from or to remote devices running Junos
```
Comment (Qalthos) [inline plugins/modules/junos_scp.py:16]:
```suggestion
short_description: (deprecated, removed after 2025-01-01) Transfer files from or to remote devices running Junos
```

File: meta/runtime.yml [hunk 4]
```diff
@@ -236,12 +236,18 @@ plugin_routing:
       redirect: junipernetworks.junos.junos_ospf_interfaces
     package:
       redirect: junipernetworks.junos.junos_package
+      deprecation:
+        removal_date: "2025-12-01"
```
Comment (Qalthos) [inline meta/runtime.yml:240]:
```suggestion
        removal_date: "2025-01-01"
```

File: meta/runtime.yml [hunk 5]
```diff
@@ -236,12 +236,18 @@ plugin_routing:
...
+        warning_text: See the plugin documentation for more details
     ping:
       redirect: junipernetworks.junos.junos_ping
     rpc:
       redirect: junipernetworks.junos.junos_rpc
     scp:
       redirect: junipernetworks.junos.junos_scp
+      deprecation:
+        removal_date: "2025-12-01"
```
Comment (Qalthos) [inline meta/runtime.yml:249]:
```suggestion
        removal_date: "2025-01-01"
```

**Review by Qalthos — CHANGES_REQUESTED** [review]
Instead reintroducing provider everywhere, Just add the provider dictionary (deprecated) to those two modules, and also modify https://github.com/ansible-collections/junipernetworks.junos/blob/main/plugins/module_utils/network/junos/junos.py#L103 to request the params from the connection plugin in addition to provider.

**Review by ashwini-mhatre — APPROVED** [review]

**Pull request merged**